polyfill/lib/ecmascript.mjs

```diff
@@ -2384,7 +2378,8 @@ export function GetNamedTimeZoneDateTimeParts(id, epochNanoseconds) {
     time: { millisecond, microsecond, nanosecond }
   } = GetISOPartsFromEpoch(epochNanoseconds);
   const { year, month, day, hour, minute, second } = GetFormatterParts(id, epochMilliseconds);
-  return BalanceISODateTime(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond);
+  const balancedTime = BalanceTime(hour, minute, second, millisecond, microsecond, nanosecond);
+  return CombineISODateAndTimeRecord({ year, month, day }, balancedTime);
```

Comment:
I'm still not sure it's correct to move away from BalanceISODateTime here. It seems like we are dropping `balancedTime.deltaDays`. (Maybe that doesn't matter because the time components are guaranteed to be balanced already in GetFormatterParts? But then why do we need to balance in the first place...)